org.osgi.test.junit5.cm/src/main/java/org/osgi/test/junit5/cm/ConfigurationExtension.java

```diff
@@ -0,0 +1,371 @@
+					.getAnnotation(WithFactoryConfiguration.class);
+				if (factoryConfigAnnotation != null) {
+					handleWithFactoryConfiguration(factoryConfigAnnotation, ca(extensionContext));
+				}
+			});
+	}
+
+	private void handleWithFactoryConfigurations(ExtensionContext extensionContext) {
+		extensionContext.getElement()
+			.ifPresent((element) -> {
+				WithFactoryConfigurations factoryConfigAnnotations = element
+					.getAnnotation(WithFactoryConfigurations.class);
+				if (factoryConfigAnnotations != null) {
+					Stream.of(factoryConfigAnnotations.value())
+						.forEachOrdered((factoryConfigAnnotation) -> {
+							handleWithFactoryConfiguration(factoryConfigAnnotation, ca(extensionContext));
+						});
+				}
+			});
+	}
+
+	private void handleWithConfiguration(ExtensionContext extensionContext) {
+		extensionContext.getElement()
+			.ifPresent((element) -> {
+				WithConfiguration configAnnotation = element.getAnnotation(WithConfiguration.class);
+				if (configAnnotation != null) {
+					handleWithConfiguration(configAnnotation, ca(extensionContext), blockingConfigHandler);
+				}
+			});
+	}
+
+	private void handleWithConfigurations(ExtensionContext extensionContext) {
+		extensionContext.getElement()
+			.ifPresent((element) -> {
+				WithConfigurations configAnnotations = element.getAnnotation(WithConfigurations.class);
+				if (configAnnotations != null) {
+					Stream.of(configAnnotations.value())
+						.forEachOrdered((configAnnotation) -> {
+							handleWithConfiguration(configAnnotation, ca(extensionContext), blockingConfigHandler);
+						});
+				}
+			});
+	}
+
+	@Override
+	public void afterEach(ExtensionContext extensionContext) throws Exception {
+		reset(extensionContext);
+
+	}
+
+	@Override
+	public void afterAll(ExtensionContext extensionContext) throws Exception {
+
+		reset(extensionContext);
+	}
+
+	private void reset(ExtensionContext extensionContext) throws Exception {
+		@SuppressWarnings("unchecked")
+		List<ConfigurationCopy> copys = getStore(extensionContext).get(STORE_CONFIGURATION_KEY, List.class);
+		ConfigUtil.resetConfig(blockingConfigHandler, ca(extensionContext), copys);
+		getStore(extensionContext).remove(STORE_CONFIGURATION_KEY, List.class);
+	}
+
+	static Store getStore(ExtensionContext extensionContext) {
+		return extensionContext
+			.getStore(Namespace.create(ConfigurationExtension.class, extensionContext.getUniqueId()));
+	}
+
+	public void updateConfigurationRespectNew(Configuration configurationToBeUpdated,
+		Dictionary<String, Object> newConfigurationProperties, boolean isNewConfiguraion)
+		throws InterruptedException, IOException {
+		if (configurationToBeUpdated != null) {
+			if (newConfigurationProperties != null
+				&& !ConfigUtil.isDictionaryWithNotSetMarker(newConfigurationProperties)) {
+				// has relevant Properties to update
+				blockingConfigHandler.update(configurationToBeUpdated, newConfigurationProperties, 1000);
+			} else if (isNewConfiguraion) {
```

Comment:
`isNewConfiguraion` -> `isNewConfiguration`. We can thank IDEs for the propagation of typos like this... :smile:

Reply:
ok